### ISSUE TITLE
huge amount of output for a single lint, weird whitespace warnings in flake8

### ISSUE THREAD

**Issue by braver** [SublimeLinter/SublimeLinter#944]
This is one pass of SublimeLinter-flake8 over style.py, after opening ST with just this file open.

Including all debug output from startup for context. The interesting bit is **15**x `wanted python is 'None'` and `trying to use globally installed flake8`,  

```
startup, version: 3143 osx x64 channel: stable
executable: /Applications/Sublime Text.app/Contents/MacOS/Sublime Text
working dir: /
packages path: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages
state path: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Local
zip path: /Applications/Sublime Text.app/Contents/MacOS/Packages
zip path: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Installed Packages
ignored_packages: ["Vintage"]
pre session restore time: 0.327635
using gpu buffer for window
startup time: 0.401545
reloading plugin Default.auto_indent_tag
reloading plugin Default.block
reloading plugin Default.comment
reloading plugin Default.convert_syntax
reloading plugin Default.copy_path
reloading plugin Default.delete_word
reloading plugin Default.detect_indentation
reloading plugin Default.duplicate_line
reloading plugin Default.echo
reloading plugin Default.exec
reloading plugin Default.fold
reloading plugin Default.font
reloading plugin Default.goto_line
reloading plugin Default.history_list
reloading plugin Default.indentation
reloading plugin Default.install_package_control
reloading plugin Default.kill_ring
reloading plugin Default.mark
reloading plugin Default.new_templates
reloading plugin Default.open_context_url
reloading plugin Default.open_in_browser
reloading plugin Default.pane
reloading plugin Default.paragraph
reloading plugin Default.paste_from_history
reloading plugin Default.profile
reloading plugin Default.quick_panel
reloading plugin Default.run_syntax_tests
reloading plugin Default.save_on_focus_lost
reloading plugin Default.scroll
reloading plugin Default.set_unsaved_view_name
reloading plugin Default.settings
reloading plugin Default.show_scope_name
reloading plugin Default.side_bar
reloading plugin Default.sort
reloading plugin Default.swap_line
reloading plugin Default.switch_file
reloading plugin Default.symbol
reloading plugin Default.transform
reloading plugin Default.transpose
reloading plugin Default.trim_trailing_white_space
reloading plugin Default.ui
reloading plugin CSS.css_completions
reloading plugin Diff.diff
reloading plugin HTML.encode_html_entities
reloading plugin HTML.html_completions
reloading plugin 0_package_control_loader.00-package_control
reloading plugin 0_package_control_loader.01-pygments
reloading plugin 0_package_control_loader.50-markupsafe
reloading plugin 0_package_control_loader.50-pymdownx
reloading plugin 0_package_control_loader.50-python-markdown
reloading plugin 0_package_control_loader.50-pyyaml
reloading plugin 0_package_control_loader.51-python-jinja2
reloading plugin 0_package_control_loader.55-jsonschema
reloading plugin 0_package_control_loader.55-mdpopups
reloading plugin A File Icon.A File Icon
reloading plugin GitGutter.plugin
reloading plugin GitSavvy.__init__
reloading plugin GitSavvy.git_savvy
reloading plugin Indent XML.indentxml
reloading plugin Indent XML.setup
reloading plugin Package Control.1_reloader
reloading plugin Package Control.2_bootstrap
reloading plugin Package Control.Package Control
reloading plugin SublimeLinter-contrib-tslint.linter
reloading plugin LESS.less_completions
reloading plugin SideBarTools.SideBar
reloading plugin SnippetMaker.SnippetMaker
reloading plugin SublimeGitUp.SublimeGitUp
reloading plugin SublimeLinter.__init__
reloading plugin SublimeLinter.busy_indicator_view
reloading plugin SublimeLinter.commands
reloading plugin SublimeLinter.goto_commands
reloading plugin SublimeLinter.highlight_view
reloading plugin SublimeLinter.panel_view
reloading plugin SublimeLinter.status_bar_view
reloading plugin SublimeLinter.sublime_linter
reloading plugin SublimeLinter-annotations.linter
reloading plugin SublimeLinter-flake8.linter
reloading plugin SublimeLinter-json.linter
reloading plugin Terminal.Terminal
plugins loaded
SublimeLinter: debug mode: on 
SublimeLinter: version: unknown 
SublimeLinter: detected syntax: python 
environment variables loaded using: /usr/local/bin/fish -l
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: SublimeLinter: flake8: wanted python is 'None' 
flake8: wanted python is 'None' 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: PATH:
    /Users/koenlageveen/Dev/utils/
    /usr/local/bin
    /usr/bin
    /bin
    /usr/sbin
    /sbin 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 

SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8 output:
    stdin:6:12: W292 no newline at end of file 
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: wanted python is 'None'SublimeLinter: flake8: wanted python is 'None'  

SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: SublimeLinter: flake8: trying to use globally installed flake8 
flake8: trying to use globally installed flake8 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8 output:
    stdin:4:12: W292 no newline at end of file 
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: flake8: wanted python is 'None' 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: trying to use globally installed flake8 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
SublimeLinter: flake8: style.py ['/usr/local/bin/flake8', '--format', 'default', '-'] 
SublimeLinter: flake8: cwd: /Users/koenlageveen/Library/Application Support/Sublime Text 3/Packages/SublimeLinter 
Package Control: Skipping automatic upgrade, last run at 2018-02-03 17:11:58, next run at 2018-02-03 18:11:58 or after

```

I check some other things, open a single python file. It appears `context_sensitive_executable_path` is sometimes hit **42** times just by opening that single file, this is not by changing anything.

What is going on here?

**Comment by kaste** [comment]
So, the requirement to reproduce this: I have one empty window, and then open one python file in it?

**Comment by braver** [comment]
I think you can just enable debug mode and then open a python file.

**Comment by kaste** [comment]
Obviously, you need to first check if `queue.hit` is called too often and very fast so that our simple `time.monotonic()` call will be constant over that time.

**Comment by kaste** [comment]
LOL, ahem, I cannot reproduce this issue (as all the other issues you had yesterday, I mean, I really opened the settings, closed that window as fast as I can. I think I clicked close, before the window was completely rendered. 😄

**Comment by braver** [comment]
Are you on Linux? 

What’s with my machine man? 😞

**Comment by kaste** [comment]
Windows for life

**Comment by braver** [comment]
It's SublimeLinter-annotations. Perhaps it's triggering the double import bug. We'll need to revise that plugin later anyway, but right now it's broken. Pfew 😅

**Comment by kaste** [comment]
Fix in #953 . Test and ship
